code/game/objects/items/stacks/stack.dm

```diff
@@ -86,46 +87,61 @@
 		. = (amount)
 
 /obj/item/stack/attack_self(mob/user)
-	interact(user)
+	list_recipes(user)
 
-/obj/item/stack/interact(mob/user)
+/obj/item/stack/proc/list_recipes(mob/user, recipes_sublist)
 	if (!recipes)
 		return
 	if (!src || get_amount() <= 0)
 		user << browse(null, "window=stack")
 		return
 	user.set_machine(src) //for correct work of onclose
+	var/list/recipe_list = recipes
+	if (recipes_sublist && recipe_list[recipes_sublist] && istype(recipe_list[recipes_sublist], /datum/stack_recipe_list))
+		var/datum/stack_recipe_list/srl = recipe_list[recipes_sublist]
+		recipe_list = srl.recipes
 	var/t1 = text("<HTML><HEAD><title>Constructions from []</title></HEAD><body><TT>Amount Left: []<br>", src, get_amount())
-	for(var/i=1;i<=recipes.len,i++)
-		var/datum/stack_recipe/R = recipes[i]
-		if (isnull(R))
+	for(var/i=1;i<=recipe_list.len,i++)
+		var/E = recipe_list[i]
+		if (isnull(E))
 			t1 += "<hr>"
 			continue
-		if (i>1 && !isnull(recipes[i-1]))
+		if (i>1 && !isnull(recipe_list[i-1]))
 			t1+="<br>"
-		var/max_multiplier = round(get_amount() / R.req_amount)
-		var/title as text
-		var/can_build = 1
-		can_build = can_build && (max_multiplier>0)
-		if (R.res_amount>1)
-			title+= "[R.res_amount]x [R.title]\s"
-		else
-			title+= "[R.title]"
-		title+= " ([R.req_amount] [singular_name]\s)"
-		if (can_build)
-			t1 += text("<A href='?src=[REF(src)];make=[];multiplier=1'>[]</A>  ", i, title)
-		else
-			t1 += text("[]", title)
-			continue
-		if (R.max_res_amount>1 && max_multiplier>1)
-			max_multiplier = min(max_multiplier, round(R.max_res_amount/R.res_amount))
-			t1 += " |"
-			var/list/multipliers = list(5,10,25)
-			for (var/n in multipliers)
-				if (max_multiplier>=n)
-					t1 += " <A href='?src=[REF(src)];make=[i];multiplier=[n]'>[n*R.res_amount]x</A>"
-			if (!(max_multiplier in multipliers))
-				t1 += " <A href='?src=[REF(src)];make=[i];multiplier=[max_multiplier]'>[max_multiplier*R.res_amount]x</A>"
+
+		if (istype(E, /datum/stack_recipe_list))
+			var/datum/stack_recipe_list/srl = E
+			if (src.amount >= srl.req_amount)
```

Comment:
is the src needed in these datums?

Reply:
Probably not but it doesn't break anything if you leave the src. in, I'll clean up the src. once it works
The issue is that in all stack recipe menus clicking the 'links' does nothing it does not produce the item